### DISCUSSION TITLE
Populate file title with (partly) creation date

### DISCUSSION THREAD

**Discussion by ashrama** [SilentVoid13/Templater#590] [Help] [answered]
Using Templater with "Trigger Templater on new file creation" enabled and "Folder templates" set.  I'm wanting to automatically populate part of the note title when the note is created.
i.e. In the file title I thought to use something like:
"My New Note <% tp.file.creation_date("YYYYMMDDHHmmss") %>"

Is there a way to include part of the file title in the template file? Or do all note file titles need to be manually specified when the note is created?

**Comment by drmikecrowe** [comment, marked as answer]
See my [gist](https://gist.github.com/drmikecrowe/f3e38008562791cc3449f482bbbbf717) for something along the same lines.  You can enhance the name with the date however you like.

**Reply by ashrama** [reply]
Thank you for this. For future reference I used the following code between the frontmatter and main note template text.

```
<%*
const note_name_text = await tp.system.prompt("Note name:");
const note_name_date = tp.file.creation_date(" YYYYMMDDHHmmss");
const note_name = note_name_text.concat(note_name_date);

console.log('Renaming note to include file creation date/time: ${note_name}');
await tp.file.rename(note_name);
%>
```
